src/hb-ot-layout.cc

```diff
@@ -1046,6 +1064,97 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
+ * @sample_id: (out) (allow-none): The ‘name’ table name ID that specifies sample text
+ *             that illustrates the effect of this feature. (May be NULL.)
+ * @num_named_parameters: (out) (allow-none):  Number of named parameters. (May be zero.)
+ * @first_param_id: (out) (allow-none): The first ‘name’ table name ID used to specify
+ *                  strings for user-interface labels for the feature
+ *                  parameters. (Must be zero if numParameters is zero.)
+ * @char_count: (out) (allow-none): The count of characters for which this feature
+ *              provides glyph variants. (May be zero.)
+ * @characters: (out) (allow-none): The Unicode Scalar Value of the characters for which
+ *              this feature provides glyph variants.
+ *
+ * Returns bunch of NameIDs and other related information for a tag,
+ * and sets NameIDs parameters (suffixed with _id) to 0 in case
+ * the requested tag not found or the tag doesn't support it.
+ *
+ * Return value: true if could find any feature with the tag, false otherwise
+ *
+ * Since: REPLACEME
+ **/
+hb_bool_t
+hb_ot_layout_get_feature_params (hb_face_t       *face,
+				 hb_tag_t         tag,
+				 unsigned int    *label_id,
+				 unsigned int    *tooltip_id,
+				 unsigned int    *sample_id,
+				 unsigned int    *num_named_parameters,
+				 unsigned int    *first_param_id,
+				 unsigned int    *char_count,
+				 hb_codepoint_t **characters)
+{
+  const OT::FeatureParams &feature_params =
+    _get_gsubgpos_matched_feature_params (face, tag);
+  if (&feature_params != &OT::Null (OT::FeatureParams))
+  {
+    const OT::FeatureParamsStylisticSet& stylistic_set_params =
+      feature_params.get_stylistic_set_params (tag);
+    if (&stylistic_set_params != &OT::Null (OT::FeatureParamsStylisticSet)) /* ssXX */
+    {
+#define PARAM(a, A) if (a) *a = A
+      PARAM(label_id, stylistic_set_params.uiNameID);
+      // ssXX features don't have the rest
+      PARAM(tooltip_id, 0);
+      PARAM(sample_id, 0);
+      PARAM(num_named_parameters, 0);
+      PARAM(first_param_id, 0);
+      PARAM(char_count, 0);
+      PARAM(characters, nullptr);
+      return true;
+    }
+    const OT::FeatureParamsCharacterVariants& character_variants_params =
+      feature_params.get_character_variants_params (tag);
+    if (&character_variants_params != &OT::Null (OT::FeatureParamsCharacterVariants)) /* cvXX */
+    {
+      PARAM(label_id, character_variants_params.featUILableNameID);
+      PARAM(tooltip_id, character_variants_params.featUITooltipTextNameID);
+      PARAM(sample_id, character_variants_params.sampleTextNameID);
+      PARAM(num_named_parameters, character_variants_params.numNamedParameters);
+      PARAM(first_param_id, character_variants_params.firstParamUILabelNameID);
+      PARAM(char_count, character_variants_params.characters.len);
+      unsigned int len = character_variants_params.characters.len;
+      if (len != 0 && characters)
```

Comment:
This should be `&& *chracters` I think.

Reply:
I think `characters` is correct but I am not sure.

Reply:
Of course it is, silly me! I need my morning tea.

Reply:
Using your tips and looking at `hb_face_get_table_tags` implementation I guess I've got it right this time, have a look please.